api/chat_main_handler.go

```diff
@@ -85,30 +83,27 @@ func (h *ChatHandler) chatHandler(w http.ResponseWriter, r *http.Request) {
 	err := json.NewDecoder(r.Body).Decode(&req)
 	if err != nil {
 		w.WriteHeader(http.StatusBadRequest)
-		fmt.Fprintf(w, "Invalid request body: %v", err)
+		json.NewEncoder(w).Encode(map[string]interface{}{"error": "Invalid request body"})
 		return
 	}
-
-	chatSessionUuid := req.SessionUuid
-	chatUuid := req.ChatUuid
-	newQuestion := req.Prompt
-	log.Printf("Received prompt: %s\n", newQuestion)
+	defer r.Body.Close()
 	ctx := r.Context()
-	userIDStr := ctx.Value(userContextKey).(string)
-	userIDInt, err := strconv.Atoi(userIDStr)
-	if err != nil {
-		http.Error(w, "Error: '"+userIDStr+"' is not a valid user ID. Please enter a valid user ID.", http.StatusBadRequest)
+	userIDStr, ok := ctx.Value(userContextKey).(string)
+	if !ok {
+		RespondWithError(w, http.StatusInternalServerError, err.Error(), err)
 		return
 	}
-	answer_msg, err := h.chatService.Chat(chatSessionUuid, chatUuid, newQuestion, int32(userIDInt))
+
+	userIDInt, _ := strconv.Atoi(userIDStr)
+	answerMsg, err := h.chatService.Chat(req.SessionUuid, req.ChatUuid, req.Prompt, int32(userIDInt))
+
 	if err != nil {
-		fmt.Fprintf(w, "problem in chat: %v", err)
+		RespondWithError(w, http.StatusInternalServerError, err.Error(), err)
 		return
 	}
 
-	// Send the response as JSON
 	w.Header().Set("Content-Type", "application/json")
-	json.NewEncoder(w).Encode(map[string]interface{}{"status": "Success", "text": answer_msg.Content, "chatUuid": answer_msg.Uuid})
+	json.NewEncoder(w).Encode(map[string]interface{}{"status": "Success", "text": answerMsg.Content, "chatUuid": answerMsg.Uuid})
 }
 
 // OpenAIChatCompletionAPIWithStreamHandler is an HTTP handler that sends the stream to the client as Server-Sent Events (SSE)
```

Comment:
:robot: OpenAI

This patch changes the error handling in the `chatHandler` function.

Instead of using `fmt.Fprintf` to write an error message to the response body, it now uses a custom `RespondWithError` function that writes a JSON-encoded error message to the response body.

This change is an improvement because it provides a more consistent and structured way of reporting errors to clients.

Please confirm that the `RespondWithError` function is defined and implemented correctly before merging this patch.

Also, please update the comments on the pull request to reflect this change.

<!-- This is an auto-generated comment by OpenAI -->